src/compiler/crystal/tools/doc/highlighter.cr

```diff
@@ -8,10 +8,7 @@ module Crystal::Doc::Highlighter
     lexer.wants_raw = true
 
     String.build do |io|
-      begin
-        highlight_normal_state lexer, io
-      rescue Crystal::SyntaxException
-      end
+      highlight_normal_state lexer, io
```

Comment:
What I mean is, the `begin/rescue` should happen in this method, not in the caller of this method.